nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -292,14 +299,47 @@ in {
 
         host = lib.mkOption {
           description = lib.mdDoc "Redis host.";
-          type = lib.types.str;
-          default = "127.0.0.1";
+          type = lib.types.nullOr lib.types.str;
+          default = if cfg.redis.createLocally && !cfg.redis.enableUnixSocket then "127.0.0.1" else null;
+            defaultText = lib.literalExpression ''
+            if config.${opt.redis.createLocally} && !config.${opt.redis.enableUnixSocket}
+            then "127.0.0.1"
+            else null
+          '';
         };
 
         port = lib.mkOption {
           description = lib.mdDoc "Redis port.";
-          type = lib.types.port;
-          default = 31637;
+          type = lib.types.nullOr lib.types.port;
```

Comment:
Could we just ignore the setting when the unix socket is used? This would make this a little less complex. Same for host.

Reply:
Fixed.